### PR TITLE
add feedback processing and forward to AI

### PR DESCRIPTION
**Pull request by rolfvdhulst**
processes feedback from robothub and forwards it to the AI (very simple for now, just passes last message along).

### CODE REVIEW

File: src/observer/filters/RobotFeedbackFilter.cpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
+//
+// Created by rolf on 29-10-21.
+//
+
+#include "observer/filters/RobotFeedbackFilter.h"
+
+void RobotFeedbackFilter::process(const proto::RobotData &data) {
+    auto& map = data.isyellow() ? lastYellowFeedback : lastBlueFeedback;
+    for(const auto& feedback : data.receivedfeedback()){
+        map[feedback.id()] = feedback;
```
Comment (emielsteerneman) [inline src/observer/filters/RobotFeedbackFilter.cpp:10]: Is this a proper copy, or will feedback (and this map[id]) go out of scope after the function returns

**Review by emielsteerneman — COMMENTED** [review]
lgtm, but I'm unsure about simply forwarding feedback to AI. Can't we use the feedback to more accurately determine the state of the robot. Also, the ballsensor might also trigger when its hugging another robot. Can a check be added that if the robot is not close to the ball, the ballsensor data is discarded?

File: src/observer/filters/RobotFeedbackFilter.cpp (hunk 1 above)
Comment (rolfvdhulst) [reply src/observer/filters/RobotFeedbackFilter.cpp:10]: This is clearly a proper copy as the maps type is proto::RobotFeedback.

**Comment by rolfvdhulst** [comment]
Yeah, we could, but I literally beuned this PR in 1 hour as you guys needed it. Using the ballsensor would require quite a bit more work.

**Review by alexander-de-ranitz — APPROVED** [review]
Looks good and worked during test

**Pull request merged**